lib/ddtrace/configuration/settings.rb

```diff
@@ -54,8 +54,14 @@ class Settings
       end
 
       option :tracer do |o|
-        o.default { Tracer.new }
-        o.lazy
+        o.default Tracer.new
```

Comment:
I'm curious on why is the Tracer not lazy anymore.

Reply:
Short answer is it actually could go either way. Prior to this PR, it wasn't lazy, so it's the safer option.

Longer answer is when a tracer is initialized, it initializes a new context provider, which in turn replaces the context on the thread local variable. This is why operating multiple tracers in an application is problematic: since they battle over this "global" state.

So I changed it to lazy, then saw tests intermittently failing. This was happening because the tests initiated their own test tracer (which set the context), then the integration under test would call for the global tracer, which would lazily instantiate then replace the context. This caused spans to become disjointed, failing tests. Thus changing to back to an eager set, with a reset block avoided that scenario.

It seems unlikely this would actually be a problem in a production environment, and we likely could rework tests instead to ensure the global tracer is instantiated before running the test. But all of this seemed more complicated and not suitable to mess with right now.